### PR TITLE
Initial work on models

### LINKED ISSUE #7: add models

### PR DESCRIPTION
**Pull request by cfeeney-va**
Fixes #7

Creating models for the SM responses.
One open question: I don't think the "category" of the message is complex enough for its own model; we can get a list of categories through their own API call but they just come in as a String with a message.  Do you have any thoughts on this @mphprogrammer ?

### CODE REVIEW

File: lib/va_healthcare_messaging/models/folder.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    attribute :unread_count, Integer
+    attribute :system_folder, Boolean
+
+    alias system_folder? system_folder
+
+    def <=>(other)
+      return -1 if folder_id < other.folder_id
+      return 1 if folder_id > other.folder_id
+      0
+    end
```
Comment (mphprogrammer) [inline lib/va_healthcare_messaging/models/folder.rb]: You could write <=> as
def <=>(other)
  folder_id <=> other.folder_id
end

File: lib/va_healthcare_messaging/models/message.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    attribute :recipient_name, String
+    attribute :read_receipt, String
+
+    alias attachment? attachment
+
+    def <=>(other)
+      return -1 if message_id < other.message_id
+      return 1 if message_id > other.message_id
+      0
+    end
```
Comment (mphprogrammer) [inline lib/va_healthcare_messaging/models/message.rb]: You could write <=> as
def <=>(other)
  message_id <=> other.message_id
end

File: lib/va_healthcare_messaging/models/triage_team.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  class TriageTeam < Base
+    attribute :triage_team_id, Integer
+    attribute :name, String
+    attribute :relation_type, String
+
+    def <=>(other)
+      return -1 if triage_team_id < other.triage_team_id
+      return 1 if triage_team_id > other.triage_team_id
+      0
+    end
```
Comment (mphprogrammer) [inline lib/va_healthcare_messaging/models/triage_team.rb]: you could write:
def <=>(other)
  triage_team_id <=> other.triage_team_id
end

**Comment by mphprogrammer** [comment]
Other than my nit-picky nature LGTM!

**Comment by mphprogrammer** [comment]
Would you like to change the <=> or would you like me to just merge?

**Comment by cfeeney-va** [comment]
I'll change it and then reassign to you :)
thanks!

**Comment by mphprogrammer** [comment]
LGTM

**Pull request merged**